### PR TITLE
cli: add "cilium bpf config list"

### PR DESCRIPTION
**Pull request by mhofstetter**
This PR introduces the possibility to list the content of the BPF map `cilium_runtime_config` via the cilium cli command `cilium bpf config list`.

In addition, these information are dumped during sysdump with the bugtool.

### CODE REVIEW

File: pkg/maps/configmap/config_map.go [hunk 1]
```diff
@@ -32,14 +32,21 @@ const (
 	AgentLiveness
```
Comment (mhofstetter) [inline pkg/maps/configmap/config_map.go:31]: @dylandreimerink @jrajahalme it looks like the "marker" entry `UsedEntries` is unused and lost its meaning. When L2 Announcement has been introduced, a new entry has been added to the end (whereas the intention of the marker entry was to be the last entry).

should we delete the entry and move `AgentLiveness` to second position?

Comment (dylandreimerink) [reply pkg/maps/configmap/config_map.go:31]: Moving `AgentLiveness` is alright with me, as long as we do it before the feature freeze. Moving after that could pose upgrade issues if the map is not recreated. I also have no idea what `UsedEntries` was supposed to be used for.

**Review by dylandreimerink — APPROVED** [review]

**Comment by mhofstetter** [comment]
removed review-request for @aditighag - as @dylandreimerink covered sig-datapath. L2Announcement (by dylan) is also using the runtime config map.

**Comment by mhofstetter** [comment]
rebased to `main` to get rid of jenkins checks :)

File: pkg/maps/configmap/config_map.go (hunk 1 above)
Comment (mhofstetter) [reply pkg/maps/configmap/config_map.go:31]: @dylandreimerink thanks - removed the constant (in agent & `common.h`)

**Review by asauber — APPROVED** [review]
Looks great. The de-duplication of the index strings is nice

**Pull request merged**